### ISSUE TITLE
Support for partitioned cookies

### ISSUE THREAD

**Issue by alexcjohnson** [pallets/flask#5295]
As part of their third-party cookie crackdown, Chrome will start requiring [partitioned cookies](https://developer.chrome.com/docs/privacy-sandbox/chips/) for content in an iframe at some point in the future. So far only [Chrome and derived browsers](https://developer.mozilla.org/en-US/docs/Web/Privacy/Partitioned_cookies#browser_compatibility) support this, but according to my testing other browsers accept and ignore the `Partitioned` attribute, so it should be safe to add.

My current goal is to future-proof a library that serves [Dash](https://dash.plotly.com/) apps in iframes and requires session cookies, and without built-in support the simplest solution I can find is to monkey-patch `werkzeug.sansio.response.dump_cookie` 🙈 

This functionality is effectively an extension of (and only relevant when) `SameSite=None` so I could imagine it being implemented as `samesite = "None; Partitioned"` (in the session cookie context, `app.config["SESSION_COOKIE_SAMESITE"] = "None; Partitioned"`) if you feel a separate attribute (`partitioned = True` and `app.config["SESSION_COOKIE_PARTITIONED"] = True`) would be excessive.

**Comment by alexcjohnson** [comment]
True, doesn't seem to have a spec, though it's pretty simple. Most of the complexity comes in how it interacts with the other attributes, which I suppose Flask could choose to either restrict or just leave up to users.

It's enabled by default since Chrome 116 after a long trial period so seems unlikely to change, but I don't see that they've stated any particular date to start enforcing it.

**Comment by davidism** [comment]
The simplicity isn't the issue here, it's the lack of a spec.
